### PR TITLE
Fix crash of linter when package in run section starts with python in nameing

### LINKED ISSUE #1231: Linter crashes under certain conditions
Issue:

The linter (version 3.6.8) crashes with:

```
File lint_recipe.py, line 485, in lintify
constraint = req.split(" ", 1)[1]
IndexError: list index out of range
```

This happens when non noarch build is triggered and the run section contains packages that start with `python` (e.g. python-dateutil) that do not have version contraints. 

Minimal example `meta.yaml`:
```
{% set name = "testapp" %}
{% set version = "0.0.1" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  - url: https://github.com/apache/{{ name }}/archive/{{ version }}.tar.gz
    sha256: 194n4140184140194124914u1n0914j1041j04104014m1401420411244210410
  - url: https://raw.githubusercontent.com/apache/{{ name }}/master/LICENSE.txt
    sha256: 832u5y3285w89y537w05w49572j5nnm151591515213532mnn141249000090914

build:
  # noarch not set
  number: 0
  script:
    - "{{ PYTHON }} -m pip install . -vv"
  skip: True  # [win or py2k]
  entry_points:
    - superset = superset.bin.superset:cli

requirements:
  build:
    - nodejs
  host:
    - python
    - pip
  run:
    - python-dateutil # without version
```

### PR DESCRIPTION
**Pull request by tim-werner**
Checklist
* [x] Added a ``news`` entry

Fixes #1231

Todo:
* [x] Add test that shows fix

### CODE REVIEW

**Comment by tim-werner** [comment]
This is an ugly fix. Fixing this in a better way requires probably some regex magic!

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -462,12 +462,12 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
                 req
                 for req in host_reqs
-                if req == str(language) or req.startswith(str(language))
+                if req == (str(language) or req.startswith(str(language))) and not req.startswith(str(language)+'-')
             ]
             filtered_run_reqs = [
                 req
                 for req in run_reqs
-                if req == str(language) or req.startswith(str(language))
+                if req == (str(language) or req.startswith(str(language))) and not req.startswith(str(language)+'-')
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]:
```suggestion
                if req == str(language) or req.startswith(str(language)+"  ")
```

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -462,12 +462,12 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
             filtered_host_reqs = [
                 req
                 for req in host_reqs
-                if req == str(language) or req.startswith(str(language))
+                if req == str(language) or req.startswith(str(language) + " ")
```
Comment (scopatz) [inline conda_smithy/lint_recipe.py]: Why not do something like `if req.partition(" ")[0] == str(language)`?

Comment (tim-werner) [reply conda_smithy/lint_recipe.py]: Thanks. That is much better

**Comment by isuruf** [comment]
Thanks. Sorry it took this long to merge.

**Pull request merged**